site/src/main/java/uk/nhs/digital/ps/beans/BaseDocument.java

```diff
@@ -63,4 +67,34 @@ protected RestrictableDate nominalPublicationDateCalendarToRestrictedDate(final
             ? RestrictableDate.restrictedDateFrom(nominalPublicationDate)
             : RestrictableDate.fullDateFrom(nominalPublicationDate);
     }
+
+    /*
+    * Converts selected geographic coverage values:
+    * If England, Wales, Scotland are checked and no others - Great Britain
+    * If England, Wales, Scotland and Northern Ireland are checked and no others - United Kingdom
+    * If England, Wales, Scotland and Northern Ireland, and Republic of Ireland are ticked - British Isles
+    * If any other combination is checked, serve up a list of the checked values
+    */
+    protected String[] geographicCoverageValuesToRegionValue(final String[] cmsValues) {
+
+        if (cmsValues == null) {
+            return null;
+        }
+
+        if (arraysMatch(cmsValues, GREAT_BRITAIN)) {
+            return new String[] {"Great Britain"};
+        } else if (arraysMatch(cmsValues, UNITED_KINGDOM)) {
+            return new String[] {"United Kingdom"};
+        } else if (arraysMatch(cmsValues, BRITISH_ISLES)) {
+            return new String[] {"British Isles"};
+        } else {
+            return cmsValues;
+        }
+    }
+
+    private boolean arraysMatch(final String[] a1, final String[] a2) {
+        Arrays.sort(a1);
```

Comment:
You don't want to sort the static lists as you could have multiple threads doing this at once. Either initialise the static arrays in the correct order or (my prefered option) have the statics as Sets and then use:

`STATIC_SET.equals(new HashSet(Arrays.asList(array)))`

as sets are inherently orderless

Reply:
Updated to use Sets